src/main/java/data/AnalysisResults.java

```diff
@@ -26,13 +26,15 @@
   private final ImmutableSet<String> categories;
   private final Optional<String> store;
   private final Optional<Long> timestamp;
+  private final Optional<Double> price;
 
-  public AnalysisResults(
-      String rawText, Set<String> categories, Optional<String> store, Optional<Long> timestamp) {
+  public AnalysisResults(String rawText, Set<String> categories, Optional<String> store,
+      Optional<Long> timestamp, Optional<Double> price) {
```

Comment:
Instead of "timestamp", can this be "purchaseTimestamp" or "transactionTimestamp". "timestamp" is unclear by itself. Change the other uses in this class to match.